Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs

```diff
@@ -78,18 +78,37 @@ private static string GetHandlerPathIfAny(string listenerUrl)
         public string XRealIp => string.IsNullOrEmpty(request.Headers["X-Real-IP"]) ? null : request.Headers["X-Real-IP"];
 
         private string remoteIp;
-        public string RemoteIp =>
-            remoteIp ??
-            (remoteIp = CheckBadChars(XForwardedFor) ??
-                        NormalizeIp(CheckBadChars(XRealIp) ??
-                         (request.RemoteEndPoint != null ? NormalizeIp(request.RemoteEndPoint.Address.ToString()) : null)));
+        public string RemoteIp
+        {
+            get
+            {
+                if (remoteIp != null)
+                {
+                    return remoteIp;
+                }
+
+                var temp = CheckBadChars(XForwardedFor);
+                if (temp.Length != 0)
+                {
+                    return remoteIp = temp.ToString();
+                }
+
+                temp = CheckBadChars(XRealIp);
+                if (temp.Length != 0)
+                {
+                    return remoteIp = NormalizeIp(temp).ToString();
+                }
+
+                return remoteIp = NormalizeIp(request.RemoteEndPoint?.Address.ToString()).ToString();
```

Comment:
what would happen if `RemoteEndPoint` is `null`, won't we get `remoteIp` set to a string of literal value "null"?
or an exception in `NormalizeIp` that is when it tries to parse IP address string that is just "null"?

Reply:
The first toString is never executed if `RemoteEndPoint` is null. And I'm fairly certain that null.ToString() would throw an exception.

Reply:
So if `RemoteEndPoint` is `null` it means this effectively transforms to
```cs
return remoteIp = NormalizeIp(null).ToString();
```
Right?

Reply:
@JustAMan Correct